openff/toolkit/typing/engines/smirnoff/parameters.py

```diff
@@ -3428,149 +3429,193 @@ def check_handler_compatibility(self, other_handler: "GBSAHandler"):
                 )
 
-            if "charge_increment" in kwargs:
-                expected_num_charge_increments = cls._expected_num_charge_increments(
-                    type_
-                )
-                num_charge_increments = len(kwargs["charge_increment"])
-                if num_charge_increments != expected_num_charge_increments:
-                    raise SMIRNOFFSpecError(
-                        f"'{type_}' virtual sites expect exactly {expected_num_charge_increments} "
-                        f"charge increments, but got {kwargs['charge_increment']} "
-                        f"(length {num_charge_increments}) instead."
-                    )
+        supports_in_plane_angle = cls._supports_in_plane_angle(type_)
+        supports_out_of_plane_angle = cls._supports_out_of_plane_angle(type_)
 
-            supports_in_plane_angle = cls._supports_in_plane_angle(type_)
-            supports_out_of_plane_angle = cls._supports_out_of_plane_angle(type_)
+        if not supports_out_of_plane_angle:
+            kwargs["outOfPlaneAngle"] = kwargs.get("outOfPlaneAngle", None)
+        if not supports_in_plane_angle:
+            kwargs["inPlaneAngle"] = kwargs.get("inPlaneAngle", None)
 
-            if not supports_out_of_plane_angle:
-                kwargs["outOfPlaneAngle"] = kwargs.get("outOfPlaneAngle", None)
-            if not supports_in_plane_angle:
-                kwargs["inPlaneAngle"] = kwargs.get("inPlaneAngle", None)
+        match = kwargs.get("match", None)
 
-            match = kwargs.get("match", None)
+        if match is None:
+            raise SMIRNOFFSpecError("the `match` keyword is missing")
 
-            if match is None:
-                raise SMIRNOFFSpecError("the `match` keyword is missing")
+        out_of_plane_angle = kwargs.get("outOfPlaneAngle", 0.0 * unit.degree)
+        is_in_plane = (
+            None
+            if not supports_out_of_plane_angle
+            else numpy.isclose(out_of_plane_angle.m_as(unit.degree), 0.0)
+        )
 
-            out_of_plane_angle = kwargs.get("outOfPlaneAngle", 0.0 * unit.degree)
-            is_in_plane = (
-                None
-                if not supports_out_of_plane_angle
-                else np.isclose(out_of_plane_angle.m_as(unit.degree), 0.0)
+        if not cls._supports_match(type_, match, is_in_plane):
+            raise SMIRNOFFSpecError(
+                f"match='{match}' not supported with type='{type_}'"
+                + ("" if is_in_plane is None else f" and is_in_plane={is_in_plane}")
             )
 
-            if not cls._supports_match(type_, match, is_in_plane):
-                raise SMIRNOFFSpecError(
-                    f"match='{match}' not supported with type='{type_}'"
-                    + ("" if is_in_plane is None else f" and is_in_plane={is_in_plane}")
-                )
+    @classmethod
+    def _supports_in_plane_angle(cls, type_: _VirtualSiteType) -> bool:
+        return type_ in {"MonovalentLonePair"}
```

Comment:
Does this make https://github.com/openforcefield/openff-toolkit/pull/1837/files#diff-fd6b852eaa3bb076240130325c7023d4041de6b0f45bee26c509e61b8a661a23R3619-R3621 redundant?

Reply:
Nice find, I think so!